tests/util.py

```diff
@@ -18,3 +23,48 @@ def override_global_tracer(tracer):
     ddtrace.tracer = tracer
     yield
     ddtrace.tracer = original_tracer
+
+
+class SnapshotFailed(Exception):
+    pass
+
+
+def snapshot(ignores=None, filename=None, snapshot_dir=None, tracer=ddtrace.tracer):
+    ignores = ignores or []
+
+    def dec(f):
+
+        @functools.wraps(f)
+        def wrapper(*args, **kwargs):
+            if len(args) > 1:
+                self = args[0]
+                clsname = self.__class__.__name__
+            else:
+                clsname = ""
+
+            module = inspect.getmodule(f)
+            token = "{}{}{}.{}".format(module.__name__, "." if clsname else "", clsname, f.__name__)
+
+            try:
+                conn = httplib.HTTPConnection(tracer.writer.api.hostname, tracer.writer.api.port)
+                conn.request("GET", "/test/start?token=%s" % token)
+                r = conn.getresponse()
+                if r.status != 200:
+                    raise SnapshotFailed("", r.read().decode())
```

Comment:
Error message?

Reply:
The error message is passed along as an argument to the exception. Although I'm not sure why I do that versus just using the message argument itself 🤔. I'll check that again